src/constructs/core/ssm.ts

```diff
@@ -0,0 +1,97 @@
+import { readFileSync } from "fs";
+import { join } from "path";
+import type { IGrantable, IPrincipal } from "@aws-cdk/aws-iam";
+import { Policy, PolicyStatement } from "@aws-cdk/aws-iam";
+import { Code, Runtime, SingletonFunction } from "@aws-cdk/aws-lambda";
+import type { Reference } from "@aws-cdk/core";
+import { Construct, CustomResource, Duration } from "@aws-cdk/core";
+import { AwsCustomResourcePolicy } from "@aws-cdk/custom-resources";
+import type { CustomResourceGetParameterProps } from "./custom-resources/interfaces";
+import type { AppIdentity } from "./identity";
+import type { GuStack } from "./stack";
+
+export interface GuSSMParameterProps {
+  parameter: string;
+  secure?: boolean;
+}
+
+export interface GuSSMIdentityParameterProps extends GuSSMParameterProps, AppIdentity {}
+
+const stripped = (str: string) => str.replace(/[-/]/g, "");
+
+export class GuSSMParameter extends Construct implements IGrantable {
+  private readonly customResource: CustomResource;
+  readonly grantPrincipal: IPrincipal;
+
+  // eslint-disable-next-line custom-rules/valid-constructors -- TODO: Remove once linting rules have been relaxed for this
+  constructor(scope: GuStack, props: GuSSMParameterProps) {
+    const { parameter } = props;
+
+    const id = (id: string) => {
+      const now = Date.now();
+      // We need to create UIDs for the resources in this construct, as otherwise CFN will not trigger the lambda on updates for resources that appear to be the same
+      const uid = now.toString().substr(now.toString().length - 4);
+      return parameter.toUpperCase().includes("TOKEN") ? `${id}-token-${uid}` : `${id}-${stripped(parameter)}-${uid}`;
+    };
```

Comment:
The comment suggests this is pretty crucial, should we explicitly test it?

Reply:
Yeah, we should. I've added two tests that hopefully capture this behaviour!

Reply:
Would be good, but not a blocker, to isolate this function to make it super obvious that given this input, we expect this output. That is, test the `id` function outside of the constructor. We can do this later as needed though.